code/game/gamemodes/miniantags/pulsedemon/pulsedemon.dm

```diff
@@ -0,0 +1,821 @@
+	for(var/mob/M in get_mobs_in_view(7, src))
+		if(M.see_invisible < invisibility)
+			continue //can't view the invisible
+		var/msg = message
+		if(self_message && M == src)
+			msg = self_message
+		M.show_message(msg, EMOTE_VISIBLE, blind_message, EMOTE_AUDIBLE)
+
+/mob/living/simple_animal/demon/pulse_demon/has_internal_radio_channel_access(mob/user, list/req_one_accesses)
+	return has_access(list(), req_one_accesses, get_all_accesses())
+
+/mob/living/simple_animal/demon/pulse_demon/proc/try_hijack_apc(obj/machinery/power/apc/A, remote = FALSE)
+	// one APC per pulse demon, one pulse demon per APC, no duplicate APCs
+	if(!is_valid_apc(A) || (A in hijacked_apcs) || apc_being_hijacked || A.being_hijacked)
+		return FALSE
+
+	to_chat(src, "<span class='notice'>You are now attempting to hijack [A], this will take approximately [hijack_time / 10] seconds.</span>")
+	apc_being_hijacked = A
+	A.being_hijacked = TRUE
+	A.update_icon()
+	if(do_after(src, hijack_time, target = A))
+		if(is_valid_apc(A))
+			finish_hijack_apc(A, remote)
+		else
+			to_chat(src, "<span class='warning'>Failed to hijack [src].</span>")
+	apc_being_hijacked = null
+	A.being_hijacked = FALSE
+	A.update_icon()
+
+// Basically this proc gives you more max charge per apc you have hijacked
+// Looks weird but it gets the job done
+/mob/living/simple_animal/demon/pulse_demon/proc/calc_maxcharge(hijacked_apcs)
+	if(!hijacked_apcs) // No APCs hijacked? No extra charge
+		return 1000
+	return 20000 * clamp(hijacked_apcs, 0, 20) + 500000 * clamp(hijacked_apcs - 20, 0, 30) + 1000000 * clamp(hijacked_apcs - 50, 0, 50) + 500000000 * max(0, hijacked_apcs - 100)
+
+/mob/living/simple_animal/demon/pulse_demon/proc/finish_hijack_apc(obj/machinery/power/apc/A, remote = FALSE)
+	var/image/apc_image = image('icons/obj/power.dmi', A, "apcemag", ABOVE_LIGHTING_LAYER, A.dir)
+	apc_image.plane = ABOVE_LIGHTING_PLANE
+	LAZYADD(apc_images[get_turf(A)], apc_image)
+	client.images += apc_image
+
+	hijacked_apcs += A
+	RegisterSignal(A, COMSIG_PARENT_QDELETING, PROC_REF(apc_deleted_handler))
+	if(!remote)
+		update_controlling_area()
+	maxcharge = calc_maxcharge(length(hijacked_apcs)) + (maxcharge - calc_maxcharge(length(hijacked_apcs) - 1))
+	to_chat(src, "<span class='notice'>Hijacking complete! You now control [length(hijacked_apcs)] APCs.</span>")
+
+/mob/living/simple_animal/demon/pulse_demon/proc/try_cross_shock(src, atom/A)
+	SIGNAL_HANDLER
+	if(!isliving(A) || is_under_tile())
+		return
+	var/mob/living/L = A
+	try_shock_mob(L)
+
+/mob/living/simple_animal/demon/pulse_demon/proc/try_shock_mob(mob/living/L, siemens_coeff = 1)
+	var/dealt = 0
+	if(current_cable && current_cable.powernet && current_cable.powernet.available_power)
+		// returns used energy, not damage dealt, but ez conversion with /20
+		dealt = electrocute_mob(L, current_cable.powernet, src, siemens_coeff) / 20
+	else if(charge >= 1000)
+		dealt = L.electrocute_act(30, src, siemens_coeff)
+		adjust_charge(-1000)
```

Comment:
Was this already so in the original PR? If I understand this correctly, it takes charge whenever you're shocking a mob. 

Personal opinion:
This'll mostly hurt the early stage of the demon, and not matter at all in the late stage of the demon, while you still get 4 shot by any EMP/ion source.

Reply:
Yeah pulse demon is kinda fucked up but I'm not screwing with the current balance in this pr

Reply:
Fair, I didn't remember it being in the original PR so I thought I'd ask for clarification anyways